tonic/models/vic/grid_params.py

```diff
@@ -665,60 +881,127 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
-        shape = (nveg_clases, ) + out_dicts['veg_dict'][var].shape[1:]
+
+        # Determine the final number of veg classes, accounting for
+        # potential new bare soil class, and determine that class's idx
+        var = 'Cv'
+        if lib_bare_idx >= 0:
+            # Bare soil class already exists at lib_bare_idx
+            extra_class = 0
+        else:
+            # Bare soil class needs to be created
+            extra_class = 1
+            lib_bare_idx = out_dicts['veg_dict'][var].shape[0]
+        nveg_classes = out_dicts['veg_dict'][var].shape[0] + extra_class
+
+        # Transfer Cv info, accounting for additional bare soil areas
+        var = 'Cv'
+        shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
         new = np.zeros(shape)
-        new[:-1, :, :] = out_dicts['veg_dict'][var]
-        new[-1, :, :] = bare
+        if extra_class:
+            new[:-1, :, :] = out_dicts['veg_dict'][var]
+        else:
+            new[:, :, :] = out_dicts['veg_dict'][var]
+        new[lib_bare_idx, :, :] += bare
+        # Ensure that Cvs sum to 1.0
         new /= new.sum(axis=0)
         new[:, ymask, xmask] = FILLVALUE_F
         out_dicts['veg_dict'][var] = new
 
-        # add dummy values for other veg vars
+        # Distribute the vegparam variables (geographically-varying)
         #   double root_depth(veg_class, root_zone, nj, ni) ;
         #   double root_fract(veg_class, root_zone, nj, ni) ;
         #   double LAI(veg_class, month, nj, ni) ;
-        for var in ['root_depth', 'root_fract', 'LAI']:
-            shape = (nveg_clases, ) + out_dicts['veg_dict'][var].shape[1:]
+        #   double fcan(veg_class, month, nj, ni) ;
+        #   double albedo(veg_class, month, nj, ni) ;
+        varnames = ['root_depth', 'root_fract']
+        if vegparam_lai and lai_src == FROM_VEGPARAM:
+            varnames.append('LAI')
+        if vegparam_fcan and fcan_src == FROM_VEGPARAM:
+            varnames.append('fcanopy')
+        if vegparam_albedo and alb_src == FROM_VEGPARAM:
+            varnames.append('albedo')
+        for var in varnames:
+            shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
             new = np.zeros(shape) + FILLVALUE_F
-            new[:-1, :, :] = out_dicts['veg_dict'][var]
-            new[-1, :, :] = 0
+            if extra_class:
+                new[:-1, :, :] = out_dicts['veg_dict'][var]
+                new[-1, :, :] = bare_vegparam[var]
+            else:
+                new[:, :, :] = out_dicts['veg_dict'][var]
+            out_dicts['veg_dict'][var] = np.ma.masked_values(new, FILLVALUE_F)
+
+        varnames = []
+        if blowing_snow:
+            varnames.append('sigma_slope')
+            varnames.append('lag_one')
+            varnames.append('fetch')
+        for var in varnames:
+            shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
+            new = np.zeros(shape) + FILLVALUE_F
```

Comment:
The better way to do this (which I just recently learned) is to use `np.full`:

```Python
new = np.full(shape, FILLVALUE_F)
```

probably worth updating all the generation of empty (prefilled) arrays in this section.